exporters/otlp/profiles/src/main/java/io/opentelemetry/exporter/otlp/profiles/ProfileDictionaryCompositor.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.otlp.profiles;
+
+import io.opentelemetry.api.common.Value;
+import io.opentelemetry.exporter.otlp.internal.data.ImmutableFunctionData;
+import io.opentelemetry.exporter.otlp.internal.data.ImmutableKeyValueAndUnitData;
+import io.opentelemetry.exporter.otlp.internal.data.ImmutableLinkData;
+import io.opentelemetry.exporter.otlp.internal.data.ImmutableLocationData;
+import io.opentelemetry.exporter.otlp.internal.data.ImmutableMappingData;
+import io.opentelemetry.exporter.otlp.internal.data.ImmutableProfileDictionaryData;
+import io.opentelemetry.exporter.otlp.internal.data.ImmutableStackData;
+import java.util.Collections;
+
+/**
+ * This class allows for the assembly of the reference tables which form a ProfileDictionaryData.
+ *
+ * <p>It's effectively a builder, though without the fluent API. Instead, mutation methods return
+ * the index of the offered element, this information being required to construct any element that
+ * references into the tables.
+ *
+ * <p>Note this class does not enforce referential integrity, i.e. will accept an element which
+ * references a non-existing element.
+ *
+ * <p>This class is not threadsafe and must be externally synchronized.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
```

Comment:
This class isn't in an 'internal' package currently, so it probably shouldn't have this notation.